### DISCUSSION TITLE
Repeatedly retrieving message UID

### DISCUSSION THREAD

**Discussion by octo-weilian** [mjs/imapclient#471] [Q&A] [answered]
Hello,

I'm developing a mail client which monitors new messages. The monitoring part is done on a thread with IDLE support and a queue. For every new message, I get an IDLE response with an EXISTS tag and message sequence number, e.g. `(5000,b'EXISTS')`. This sequence number is put in a queue. A second thread consumes any new sequence number from the queue and retrieve the message UID. See part of the code below. 

Let say I receive about 100 new messages in a burst. My client will be repeatedly retrieving the UID via `search()`. I'm troubled about how the server might disconnects me. Is repeatedly retrieving UID taxing for an IMAP server? If so how can I prevent this?

PS: It is important for me to use message UID due to its persistency across clients. 

```
#runs on a second thread
def get_uid(queue):
   client = IMAPClient(server,port,use_uid=True)
   login = client.login(user,secret)
   inbox = client .select_folder("INBOX",readonly=True)

   while True:
        sequence= queue.get()
        uid = client.search( str(sequence) )
        queue.task_done()
```

**Comment by mjs** [comment, marked as answer]
Unfortunately, the `EXISTS` response isn't giving you a message sequence number, just the number of messages in the folder so you can't use the number sent with `EXISTS` the way you are.

In my experience it's best to just use any IDLE response as a signal that something changed and then use some other mechanism to figure out what changed. There is too much variation between servers to bother with interpreting specific IDLE responses.

I would suggest that given that UIDs always increase and should persist between sessions, you should track the highest previously seen UID in the folder and then use something like `SEARCH UID 99:*` (where 99 is the last seen highest UID for the folder) when an IDLE response is received to check for new messages.

To really to do this right, please also read about `UIDVALIDITY` in RFC9051.
